### ISSUE TITLE
[scheduler_perf] Transient API error aborts the scheduler perf test

### ISSUE THREAD

**Issue by Huang-Wei** [kubernetes/kubernetes#96651]
In recent testings using [scheduler_perf](https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf), I encountered the following error which aborts the entire test suite:

```
--- FAIL: BenchmarkPerfScheduling/SchedulingPodAntiAffinity/500Nodes
    scheduler_perf_test.go:366: op 2: Error creating pod: Failed to create object with non-retriable error: The POST operation against Pod could not be completed at this time, please try again.
```

The source comes from API machinery code:

https://github.com/kubernetes/kubernetes/blob/6dddea5abfa20a22e93114d66735441f04f4a39c/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L314-L329

According to the comment, upon this transient API error, we should retry the request instead of quit and abort the whole test.

/sig scheduling
/priority important-longterm

**Comment by liggitt** [comment]
cc @lavalamp 

is this a response produced by priority & fairness kicking in?

**Comment by lavalamp** [comment]
It's possible, we can investigate.

**Comment by adtac** [comment]
No, this has existed for multiple months. I wrote the current version of scheduler_perf and I remember seeing this error way back in August/September.

**Comment by yue9944882** [comment]
mmm the ap&f filter doesnt actually return such error as `NewServerTimeout(..)`, the failing request seems to be exceeding  the global request time-out instead ap&f queuing timeout. additionally the rejected requests by the filter can be tracked/confirmed by the `apiserver_flowcontrol_rejected_requests_total{reason="time-out"}` metrics at the kube-apiserver.

> https://kubernetes.io/docs/concepts/cluster-administration/flow-control/#metrics
> time-out, indicating that the request was still in the queue when its queuing time limit expired.

**Comment by liggitt** [comment]
Sounds good, thanks for checking

**Comment by tkashem** [comment]
@Huang-Wei https://github.com/kubernetes/kubernetes/pull/96061 merged three days ago.Although  `NewServerTimeout(..)` is not thrown by request timeout logic. You will see a `504` from the server with the following message:
- https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go#L56
- https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go#L267

It just makes me curious with the timing of the test failure and the PR merge.

**Comment by lavalamp** [comment]
Thanks @yue9944882, I didn't think it was P&F but I wasn't certain we failed too-long requests via a different code path.

I think there's still a way we could have made this slightly worse, though, imagine a request sat in our queue for the max of 15 seconds. Imagine this request will take 55 seconds to complete. From the perspective of the global time out, it will take 55+15 = 70 seconds and be failed with this message (70 > 60). Without APF it theoretically might have succeeded, if nothing else changed.

...However, we expect a server that's taking that long to process a request is probably timing out random requests and is not actually that healthy, so I wouldn't really consider this a bug unless we've got requests that take more than 45 seconds for an unburdened control plane to process (we really should not).

**Comment by alculquicondor** [comment]
any chance we can fix this before the test freeze?

**Comment by MikeSpreitzer** [comment]
Are we still seeing a problem?  If so, an example CI failure would be helpful to examine.

**Comment by alculquicondor** [comment]
@Huang-Wei can you confirm if this is still happening?

**Comment by Huang-Wei** [comment]
I haven't run perf test in large scale recently. But I'm not aware of recent PRs fixing this behavior, so I suppose this problem still exist.
